Solutions/Zscaler Private Access (ZPA)/Analytic Rules/ZscalerSharedZPASession.yaml

```diff
@@ -0,0 +1,53 @@
+id: 40a98355-0e52-479f-8c91-4ab659cba878
+name: Zscaler - Shared ZPA session
+description: |
+  'Detects Shared ZPA session.'
+severity: Medium
+requiredDataConnectors:
+  - connectorId: ZscalerPrivateAccess
+    dataTypes:
+      - ZPAEvent
+queryFrequency: 1h
+queryPeriod: 1h
+triggerOperator: gt
+triggerThreshold: 0
+tactics:
+  - InitialAccess
+relevantTechniques:
+  - T1178
+  - T1133
+query: |
+  let open_sessions =
+  ZPAEvent
+  | where DvcAction == 'open'
+  | summarize timestampstart = min(TimeGenerated) by DstUserName, SrcIpAddr, NetworkSessionId
+  | sort by timestampstart asc;
+  let closed_sessions =
+  ZPAEvent
+  | where DvcAction == 'close'
+  | summarize timestampend = max(TimeGenerated) by DstUserName, SrcIpAddr, NetworkSessionId
+  | sort by timestampend asc;
+  open_sessions
+  | join (closed_sessions) on DstUserName, SrcIpAddr, NetworkSessionId

```

Comment:
The detection will work only if both open and close are within the same 1 hour running period. Any session, even if short, that happens to start before the one hour mark, will be ignored. The query has to look much further back for the open action.

Reply:
- Thanks for the update. The query will now run every one hour and will find overlaps for 24 hours back, creating duplicates. You need to check for "close" just in the last hour, while open can span the 24 hours.
- Joining by NeworkSessoinId and SrcIp mean that only open and close for the same session are joined (and always with the same IP) and not for different sessions. This will imply that the detection will not work. I think you want to join only on User.

Reply:
> > > > @vmanojreddy : while the section is marked at outdated and I see a new commit titled "fix" when looking at the file I don't see any changes.
> > > 
> > > 
> > > @spsocprime Can you please make the relevant changes. Thanks.
> > 
> > 
> > As I've already mentioned in comments above: <img alt="image" width="883" src="https://user-images.githubusercontent.com/94110440/157192710-10fcbbea-c030-43cb-97d1-9179f1054c55.png">
> 
> Hey @oshezaf, could you please review and approve the changes addressed by author, thanks.

This is fixed, I approved and merged.